src/metabase/events/schema.clj

```diff
@@ -206,23 +207,34 @@
    [:user-id [:maybe pos-int?]]
    [:model [:or :keyword :string]]])
 
-(mr/def ::data-editing-events
+(mr/def ::nano-id ::common/non-blank-string)
+
+(mr/def ::action-events
   [:map #_{:closed true}
-   (-> [:table_id pos-int?] (with-hydrate :table table-hydrate))
+   [:action :keyword]
+   ;; TODO ... this shouldn't be snake
+   [:invocation_id ::nano-id]
+   ;; TODO well, some can have it :/ not feeling great about defining non-generic fields here
+   (-> [:table_id {:optional true} pos-int?] (with-hydrate :table table-hydrate))
```

Comment:
yea, I don't think we should include optional keys at top level here, why do you need it?

Reply:
I'm just preserving it, as you had it in the bespoke `data-editing` events.

I guess we could move things down a bit on extract the value only when we're writing to the database, but then we have to deal with the fact that it might be in the `args`, `result`, or `info` field depending on what lifecycle event it is.

I purposely chose different key names for these, to make it harder to accidentally handle an error as a success etc, but maybe it's not worth it.